cdap-app-templates/cdap-etl/hydrator-spark-core-base/src/main/java/io/cdap/cdap/etl/spark/batch/relation/SparkSQLRelation.java

```diff
@@ -136,15 +140,17 @@ public List<String> getColumns() {
   }
 
   private String generateSelectQuery(String filterCondition) {
-    return generateSelectQuery(filterCondition, generateColumnExpMap(this.columns));
+    return generateNestedSelect(filterCondition, generateColumnExpMap(this.columns));
   }
 
   private String generateSelectQuery(String filterCondition, Map<String, Expression> columnExpMap) {
     StringBuilder queryBuilder = new StringBuilder(
-      String.format("%s %s %s %s",
+      String.format("%s %s %s %s%s%s",
```

Comment:
This is done intentionally because on line number 49, the AS variable has space padding on both sides which is used in getColumnAliasCSV (). Also if we add additional spaces to a SQL query, it throws syntax errors.

Reply:
Let's make 
`AS = "AS";` i.e without spaces and add spaces in Formatting. That's more readable